### PR TITLE
Detect empty 'require' and ':require' forms

### LINKED ISSUE #2416: Empty require in ns doesn't throw an error
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2024.08.29

**macro usage**

N/A

**platform**

All

**editor**

N/A

**problem**

```clojure
(ns foo (:require))
```

isn't an error.

**repro**

```
$ echo "(ns foo (:require))" | clj-kondo --lint -
linting took 6ms, errors: 0, warnings: 0
```

**config**

N/A

**expected behavior**

```
$ echo "(ns foo (:require))" | clj-kondo --lint -
<stdin>:1:19: error: require form is invalid: form must not be empty
linting took 5ms, errors: 1, warnings: 0
```

If this is accepted, I can write the PR to add it.

### PR DESCRIPTION
**Pull request by NoahTheDuke**
Please answer the following questions and leave the below in as part of your PR.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

- [x] I have read the [developer documentation](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [x] This PR contains a [test](https://github.com/clj-kondo/clj-kondo/blob/master/doc/dev.md#tests) to prevent against future regressions

- [x] I have updated the [CHANGELOG.md](https://github.com/clj-kondo/clj-kondo/blob/master/CHANGELOG.md) file with a description of the addressed issue.

Closes #2416

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
I narrowed it to just Clojure, instead of excluding clojurescript. I don't know if that's the appropriate move, but seemed better to be safe.

File: src/clj_kondo/impl/analyzer/namespace.clj [hunk 1]
```diff
@@ -369,6 +369,11 @@
         unused-namespace-disabled? (identical? :off (-> ctx :config :linters :unused-namespace :level))
         analyzed
         (map (fn [[require-kw libspecs]]
+               (when (identical? :clj lang)
```
Comment (borkdude) [inline src/clj_kondo/impl/analyzer/namespace.clj:372]: yeah, let's just warn for every language, also for CLJS

**Comment by borkdude** [comment]
Somehow comments as a part of an outdated line of code never come through, but here's a screenshot:

Screenshot 2024-10-30 at 20 00 18

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**